lisa/sut_orchestrator/azure/common.py

```diff
@@ -171,80 +172,78 @@ def __post_init__(self, *args: Any, **kwargs: Any) -> None:
 
     @property
     def marketplace(self) -> Optional[AzureVmMarketplaceSchema]:
-        # this is a safe guard and prevent mypy error on typing
-        if not hasattr(self, "_marketplace"):
-            self._marketplace: Optional[AzureVmMarketplaceSchema] = None
-        marketplace: Optional[AzureVmMarketplaceSchema] = self._marketplace
-        if not marketplace:
-            if isinstance(self.marketplace_raw, dict):
+
+        if self._marketplace is not None:
```

Comment:
The marketplace object is used to make sure the `self._marketplace` is a final state, instead of an intermedia state. The value may be run at the same time. There is a race condition to get incomplete object. So it needs the local variable `marketplace` to get the whole object set correctly, then assign to `self._marketplace`